python/cudf/cudf/io/parquet.py

```diff
@@ -953,3 +844,24 @@ def __enter__(self):
 
     def __exit__(self, *args):
         self.close()
+
+
+def _default_open_file_options(open_file_options, columns, row_groups):
+    # Copy and update `open_file_options` to include
+    # column and row-group information under the
+    # "precache_options" key. By default, we set
+    # "method" to "parquet", but precaching will
+    # be disabled if the user chooses `method=None`
+    open_file_options = (open_file_options or {}).copy()
+    precache_options = open_file_options.pop("precache_options", {}).copy()
+    if precache_options.get("method", "parquet") == "parquet":
+        precache_options.update(
+            {
+                "method": "parquet",
+                "engine": precache_options.get("engine", "pyarrow"),
+                "columns": columns,
+                "row_groups": row_groups,
+            }
+        )
+    open_file_options["precache_options"] = precache_options
+    return open_file_options
```

Comment:
It seems like in the case that `open_file_options` is an empty dict there's no need to puzzle through anything else within the dict and rather we could just return a constant preset dict that is computed at import time

Reply:
I don;t think there is any way to define a preset dict at import time.  In the case that `open_file_options` is `None` or `{}`, we need to populate `precache_options` with the specific set of `columns` and `row_groups` being accessed in the following read operation.